reservations/flightreservation/flightreservation.cpp

```diff
@@ -20,14 +20,26 @@
 #include "flightreservation.h"
 #include "src/singletonfactory.h"
 #include "src/datamap.h"
-#include <KCoreAddons/KPluginFactory>
 
 #include <QtCore/QDateTime>
 #include <QtCore/QDebug>
+#include <KCoreAddons/KPluginFactory>
+
+K_PLUGIN_FACTORY_WITH_JSON( KdeNowPluginFactory,
```

Comment:
This is done so that one can instantiate multiple objects of plugin at one time. I'm not sure if it's needed in this case or not. Anyway it's upto you.